digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataExporter/AssetReportCsvExporter.scala

```diff
@@ -0,0 +1,264 @@
+      case _ => ""
+    }
+  }
+
+
+  def extractRecords( exportAssetReports: List[ExportAssetReport], withTransaction: Boolean = true): String = {
+
+    def getTotalNewLawTrafficSigns(municipalityName: String): Int = {
+      if(withTransaction) {
+        withDynSession {
+          val municipalityInfo = municipalityDao.getMunicipalityIdByName(municipalityName)
+          assetReporterDAO.getTotalTrafficSignNewLaw(municipalityInfo.head.id)
+        }
+      }
+      else{
+        val municipalityInfo = municipalityDao.getMunicipalityIdByName(municipalityName)
+        assetReporterDAO.getTotalTrafficSignNewLaw(municipalityInfo.head.id)
+      }
+    }
+
+
+    val output = if (exportAssetReports.isEmpty) {
+        List()
+      }
+      else {
+        val allInfoGroupedByMunicipality = exportAssetReports.groupBy(_.municipalityName)
+
+        val result = allInfoGroupedByMunicipality.map { case (municipality, records) =>
+          val line = municipality + ";\r\n"
+
+          val rows = records.map { elem =>
+                val (totalPoints, isLinear) = if (elem.assetGeomType.trim == "linear") ("", "Kyllä")
+                                              else (elem.totalAssets, "")
+
+                val operatorDate = extractDateToString(elem.operatorModifiedDate)
+                val municipalityDate = extractDateToString(elem.municipalityModifiedDate)
+
+                val baseRow = s";${elem.assetNameFI};$totalPoints;$isLinear;$operatorDate;$municipalityDate;${elem.operatorUser};${elem.municipalityUser}"
+
+                if (elem.assetTypeId == TrafficSigns.typeId) {
+                  val totalNewLawTrafficSigns = getTotalNewLawTrafficSigns(municipality)
+                  baseRow.concat(s";$totalNewLawTrafficSigns")
+                }
+                else
+                  baseRow
+                }
+
+          line + rows.mkString("\r\n")
+        }
+
+      result.toList
+    }
+
+    output.mkString("\r\n")
+  }
+
+
+  def decodeAssetsToProcess(assetTypesList: List[Int]): List[Int] = {
+
+    val ( pointAssets, linearAssets ) = ( assetTypesList.contains(PointAssets.id), assetTypesList.contains(LinearAssets.id) )
+
+    if ( assetTypesList.contains(AllAssets.id) ) {
+      AssetTypeInfo.values.map(_.typeId).toList
+    }
+    else if ( pointAssets || linearAssets ) {
+
+      val (linearAssetType, pointAssetType) = AssetTypeInfo.values.partition( _.geometryType == "linear" )
+
+      val allPoints = pointAssetType.map(_.typeId)
+      val allLinears = linearAssetType.map(_.typeId)
+
+      (pointAssets, linearAssets) match {
+        case (true, false) =>
+          (allPoints ++ assetTypesList).filterNot(_ == PointAssets.id).toList
+
+        case (false, true) =>
+          (allLinears ++ assetTypesList).filterNot(_ == LinearAssets.id).toList
+
+        case (true, true) =>
+          (allPoints ++ allLinears).filterNot(id => id == PointAssets.id || id == LinearAssets.id).toList
```

Comment:
(allPoints ++ allLinears) should be enough since you are not adding the assetTypesList so the ids 1/2 will not appear

Reply:
I disagree with you @FilPinto.
If we reach this point it's because the user sent 2 and 3 in assetTypesList List, if that is possible we will enter here and we will get the points and the linears and we need to exclude those dummy values from the list.

What we can do better is ignore the Case(true, true) and do something like this:
```Scala
      (pointAssets, linearAssets) match {
        case (true, false) =>
          (allPoints ++ assetTypesList).filterNot(_ == PointAssets.id).toList

        case (false, true) =>
          (allLinears ++ assetTypesList).filterNot(_ == LinearAssets.id).toList

        case _ =>
          AssetTypeInfo.values.map(_.typeId).toList
      }

Reply:
done